behaviortree_ros2/include/behaviortree_ros2/bt_topic_sub_node.hpp

```diff
@@ -173,6 +175,16 @@ class RosTopicSubNode : public BT::ConditionNode
    */
   virtual NodeStatus onTick(const std::shared_ptr<TopicT>& last_msg) = 0;
 
+  /** Clear the message that has been processed. If returns true and no new message is 
+   * received, before next call there will be no message to process. If returns false,
+   * the next call will process the same message again, if no new message received.
+   * 
+   * This can be equated with latched vs non-latched topics in ros 1.
+   * 
+   * @return true will clear the message after ticking/processing.
+   */
+  virtual bool clearProcessedMessage() { return true; }
```

Comment:
I don't understand why by default this should return true. Can you elaborate? When should a user override this?

Reply:
That retains current behaviour of the node. Anything else and current implementation behaviour will change